tests/utils/test_client.py

```diff
@@ -54,3 +55,31 @@ def test_client_enforce_healthy():
     assert my_client.protocol == CONFIG["my_instance_name"]["protocol"]
     assert my_client.base_path == "/api/versioned/v1/"
     assert my_client.auth == UsernamePasswordAuth("admin", os.environ["TAMR_TOOLBOX_PASSWORD"],)
+
+
+@mock_api()
+def test_get_with_connection_retry():
+
+    tempdir = tempfile.TemporaryDirectory()
+
+    log_prefix = "caught_connection_error"
+    log_file_path = os.path.join(tempdir.name, f"{log_prefix}_{utils.logger._get_log_filename()}")
+
+    my_client = utils.client.create(**CONFIG["my_other_instance"])
+    utils.logger.enable_toolbox_logging(
+        log_to_terminal=False, log_directory=tempdir.name, log_prefix=log_prefix
+    )
+
+    with pytest.raises(TimeoutError):
+        utils.client.get_with_connection_retry(
+            my_client, "/api/service/health", timeout_seconds=10, sleep_seconds=1
+        )
+
+    with open(log_file_path, "r") as f:
+        # confirm that the intended warning was written to the log
+        assert "Caught exception in connect" in f.read()
+
+    try:
+        tempdir.cleanup()
+    except (PermissionError, NotADirectoryError):
+        tempdir.cleanup()
```

Comment:
Please add comments here so it's clear why the try-except is needed.
```suggestion
    # Cleanup temp directory
    try:
        tempdir.cleanup()
    except (PermissionError, NotADirectoryError):
        # Windows sometimes fails so try one more time
        tempdir.cleanup()
```